include/rocksdb/slice.h

```diff
@@ -179,8 +179,8 @@ class PinnableSlice : public Slice, public Cleanable {
     }
   }
 
-  void remove_prefix(size_t /*n*/) {
-    assert(0);  // Not implemented
```

Comment:
we use `assert` in a lot of places. what warning does this particular one cause?